pkg/server/endpoints/harvester.go

```diff
@@ -43,13 +47,108 @@ func NewHarvesterAPIHandlers(l logrus.FieldLogger, ds datastore.Datastore, jwtIs
 	}
 }
 
-// GetRelationships list all the relationships - (GET /relationships)
-func (h *HarvesterAPIHandlers) GetRelationships(ctx echo.Context, params harvester.GetRelationshipsParams) error {
-	return nil
+// GetRelationships list all the relationships for a given trust domain name and consent status  - (GET /relationships)
+// The consent status is optional, if not provided, all relationships will be returned for a given trust domain. If the
+// consent status is provided, only relationships with the given consent status for the given trust domain will be returned.
+// The trust domain name provided should match the authenticated trust domain.
+func (h *HarvesterAPIHandlers) GetRelationships(echoCtx echo.Context, params harvester.GetRelationshipsParams) error {
+	ctx := echoCtx.Request().Context()
+
+	authTD, err := h.getAuthenticateTrustDomain(echoCtx, params.TrustDomainName)
+	if err != nil {
+		return err
+	}
+
+	consentStatus := *params.ConsentStatus
+
+	switch consentStatus {
+	case "", api.Accepted, api.Denied, api.Pending:
+	default:
+		err := fmt.Errorf("invalid consent status: %q", *params.ConsentStatus)
+		return h.handleErrorAndLog(err, err.Error(), http.StatusBadRequest)
+	}
+
+	// get the relationships for the trust domain
+	relationships, err := h.Datastore.FindRelationshipsByTrustDomainID(ctx, authTD.ID.UUID)
+	if err != nil {
+		msg := "error looking up relationships"
+		err := fmt.Errorf("%s: %w", msg, err)
+		return h.handleErrorAndLog(err, msg, http.StatusInternalServerError)
+	}
+
+	if consentStatus != "" {
+		relationships = filterRelationshipsByConsentStatus(authTD.ID.UUID, relationships, consentStatus)
+	}
+
+	apiRelationships := make([]*api.Relationship, 0, len(relationships))
+	for _, r := range relationships {
+		apiRelationships = append(apiRelationships, api.RelationshipFromEntity(r))
+	}
+
+	return chttp.WriteResponse(echoCtx, apiRelationships)
 }
 
-// PatchRelationshipsRelationshipID accept/denied relationships requests - (PATCH /relationships/{relationshipID})
-func (h *HarvesterAPIHandlers) PatchRelationshipsRelationshipID(ctx echo.Context, relationshipID api.UUID) error {
+// PatchRelationship accept/denied relationships requests - (PATCH /relationships/{relationshipID})
```

Comment:
super ultra epic picky nit
```suggestion
// PatchRelationship accept/denies relationships requests - (PATCH /relationships/{relationshipID})
```